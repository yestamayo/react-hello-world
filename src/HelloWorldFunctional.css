.HelloWorldFunctional {
  border: 2px solid black;
  text-align: center;
  background: #CCC;
  color: black;
  margin: 20px;
  padding: 20px;
}