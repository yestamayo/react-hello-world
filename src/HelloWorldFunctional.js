import React, {Component} from 'react';
import './HelloWorldFunctional.css';

class HelloWorldFunctional extends Component{

    constructor(props){
        super(props);
        this.state = { 
            greeting: 'Hola' 
        };
        this.frenchify = this.frenchify.bind(this);
        this.removeGreeting = this.removeGreeting.bind(this);
    }

    frenchify() {
        this.setState({ greeting: 'Bonjour' });
    }

    render() {
        return (
            <div className="HelloWorldFunctional">
                {this.state.greeting} {this.props.name}!
                <br/>
                <button className="frenchify" onClick={this.frenchify}>Frenchify!</button>
                <br/>
                <button className="remove" onClick={this.removeGreeting}>Remove Me!</button>
            </div>
        );
    }

    removeGreeting() {
        this.props.removeGreeting(this.props.name);
    }
}

export default HelloWorldFunctional;
