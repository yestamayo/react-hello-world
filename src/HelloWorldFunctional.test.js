import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import HelloWorldFunctional from './HelloWorldFunctional';

describe(HelloWorldFunctional, () => {
  // Add the rest of our tests here later!
});

const name = 'Person';
const mockRemoveGreeting = jest.fn();
const component = shallow(
  <HelloWorldFunctional name={name} removeGreeting={mockRemoveGreeting}/>
);

it('renders and matches our snapshot', () => {
  const component = renderer.create(
    <HelloWorldFunctional name="Person" />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('contains the supplied name', () => {
  expect(component.text()).toContain(name);
});

it('modifies the greeting when frenchify button is clicked', () => {
  component.find('button.frenchify').simulate('click');
  expect(component.text()).toContain('Bonjour');
});

it('calls the passed in removeGreeting function when remove button is clicked', () => {
  component.find('button.remove').simulate('click');
  expect(mockRemoveGreeting).toBeCalled();
});