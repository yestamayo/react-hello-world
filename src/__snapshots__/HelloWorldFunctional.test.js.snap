// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders and matches our snapshot 1`] = `
<div
  className="HelloWorldFunctional"
>
  Hola
   
  Person
  !
  <br />
  <button
    className="frenchify"
    onClick={[Function]}
  >
    Frenchify!
  </button>
  <br />
  <button
    className="remove"
    onClick={[Function]}
  >
    Remove Me!
  </button>
</div>
`;
